components/eamxx/src/share/field/field_alloc_prop.hpp

```diff
@@ -61,21 +61,37 @@ namespace scream
 
 // Helper struct to store info related to the subviewing process
 struct SubviewInfo {
-  SubviewInfo () = default;
-  SubviewInfo (const int dim, const int slice, const int extent, const bool is_dynamic) :
-    dim_idx(dim), slice_idx(slice), dim_extent(extent), dynamic(is_dynamic) {}
-  SubviewInfo (const SubviewInfo&) = default;
-  SubviewInfo& operator= (const SubviewInfo&) = default;
-
-  int dim_idx    = -1;  // Dimension along which slicing happened
-  int slice_idx  = -1;  // Slice along dimension $dim_idx
-  int dim_extent = -1;  // Extent of dimension $dim_idx
-  bool dynamic   = false; // Whether this is a dynamic subview (slice_idx can change)
+  SubviewInfo() = default;
+  SubviewInfo(const int dim, const int slice, const int extent,
+              const bool is_dynamic)
+      : dim_idx(dim), slice_idx(slice), dim_extent(extent),
+        dynamic(is_dynamic) {}
+  // multi-slice subview across contiguous indices
+  SubviewInfo(const int dim, const int slice_beg, const int slice_end,
+              const int extent, const bool is_dynamic)
+      : dim_idx(dim), slice_idx(slice_beg), slice_idx_end(slice_end),
+        dim_extent(extent), dynamic(is_dynamic) {}
+  SubviewInfo(const SubviewInfo&) = default;
+  SubviewInfo& operator=(const SubviewInfo&) = default;
+
+  int dim_idx = -1;    // Dimension along which slicing happened
+  // Slice along dimension $dim_idx if taking single-index slice for subfield
+  // If taking multi-slice subfield, then this is the starting slice index
+  // e.g., slicing (:, slice_idx_end : slice_idx_end, :)
+  int slice_idx = -1;
+  // beginning slice index for multi-slice (remains == -1 if single-slice subview)
+  int slice_idx_beg = -1;
+  // ending slice index for multi-slice (remains == -1 if single-slice subview)
+  int slice_idx_end = -1;
+  int dim_extent = -1; // Extent of dimension $dim_idx
+  bool dynamic =
+      false; // Whether this is a dynamic subview (slice_idx can change)
 };
 
 inline bool operator== (const SubviewInfo& lhs, const SubviewInfo& rhs) {
   return lhs.dim_idx==rhs.dim_idx &&
          lhs.slice_idx==rhs.slice_idx &&
+        //  FIXME: change for multi-slice (throw error to check?)
```

Comment:
I think we need to fix this check before we merge.

Reply:
done!